### PR TITLE
Logging standardization: standardize logger declarations, parameterize log calls, fix stack traces, replace println, add outcome logging

### LINKED ISSUE #126: As a registry operator, I want per-product outcome logging (SUCCESS / FAILED / SKIPPED) so I can determine what happened to any specific product by LIDVID
## Summary

As a registry operator, I want the registry-loader to emit a clear per-product outcome log line (SUCCESS, FAILED, or SKIPPED) containing the product's LIDVID, so that I can quickly determine what happened to any specific product during a harvest or load run.

## Motivation

The tool's core job is to load or update products in the registry. Currently, logging is operation-centric (bulk API calls, schema updates) rather than product-centric. An operator has no reliable way to grep logs and answer: "Was LIDVID `urn:nasa:pds:foo::1.0` loaded successfully?"

## Acceptance Criteria

- Every product processed emits exactly one outcome log line containing its LIDVID and one of three states:
  - `[SUCCESS]` — product was written to the registry (new insert or update)
  - `[FAILED]`  — product was attempted but could not be written (include reason)
  - `[SKIPPED]` — product was intentionally not loaded (include reason, e.g. version policy)
- Log format is consistent and grep-friendly, e.g.:
  ```
  [SUCCESS] lid=urn:nasa:pds:... vid=1.0
  [FAILED]  lid=urn:nasa:pds:... vid=1.0  reason=<message>
  [SKIPPED] lid=urn:nasa:pds:... vid=1.0  reason=already loaded with newer version
  ```
- The outcome line is emitted at a level visible to operators by default (e.g., the existing `SUMMARY` custom level or INFO)
- Full exception stack traces for `FAILED` cases go to DEBUG level — operators should not have to wade through stack traces to see what failed
- Both harvest and manager emit outcome lines using the same format

## Notes

- The harvest module already defines a custom `SUMMARY` log level (`LogUtils.LEVEL_SUMMARY`, priority 150). This should be evaluated as the appropriate level for outcome lines, or promoted/renamed as part of this work.
- This task is the highest-priority item in the logging standardization theme.

---
## 🎉 Integration & Test

> 🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### LINKED ISSUE #127: Fix exception logging to always include stack traces
## Summary

Several error-handling sites in harvest and common log only the exception message string, silently dropping the full stack trace. This makes it impossible to diagnose the root cause of failures from logs alone.

## Affected Files

- `harvest/src/main/java/gov/nasa/pds/harvest/crawler/FilesProcessor.java` lines 153, 172 — `log.error(ex.getMessage())`
- `harvest/src/main/java/gov/nasa/pds/harvest/crawler/ProductProcessor.java` lines 146, 168 — `log.warn/error(ex.getMessage())`
- `common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java` line 77 — `log.error("OSE message: {} <status code = {}>", ose.getMessage(), ose.status())`

## Acceptance Criteria

- All exception log calls pass the exception object as the final argument so the full stack trace is preserved:
  - `log.error("message", ex)` not `log.error(ex.getMessage())`
- Stack traces appear at the same level as the message (not demoted to DEBUG) unless the issue is genuinely expected/non-fatal, in which case the rationale should be commented
- No regression: existing correct sites (e.g., `HarvestCli.java`, `SchemaUpdater.java`) remain unchanged

---
## 🎉 Integration & Test

> 🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### LINKED ISSUE #128: Consolidate Log4jConfigurator into common module and resolve DEBUG vs ALL divergence
## Summary

The `Log4jConfigurator` class is duplicated in both the harvest and manager modules. The two copies have diverged: requesting `"DEBUG"` verbosity produces `Level.DEBUG` in harvest but `Level.ALL` in manager — a silent behavioral difference for operators. Both copies should be consolidated into the `common` module.

## Details

- `harvest/src/main/java/gov/nasa/pds/harvest/util/log/Log4jConfigurator.java`
- `manager/src/main/java/gov/nasa/pds/registry/mgr/util/log/Log4jConfigurator.java`

The only intentional differences between the two are the default log file name (`harvest.log` vs `registry-manager.log`) and the `DEBUG`→`Level.ALL` divergence. Everything else is identical.

The harvest module also defines a custom `SUMMARY` log level in `LogUtils.java` that manager does not have, creating an inconsistent operator experience.

## Acceptance Criteria

- A single `Log4jConfigurator` lives in `common` and is used by both harvest and manager
- Default log file name is passed as a parameter (or each CLI passes its own default)
- The `DEBUG` vs `ALL` divergence is resolved with a single consistent behavior
- The `SUMMARY` custom level (or equivalent) is promoted to `common` so both tools can use it
- Both harvest and manager CLI tools continue to work identically from an operator perspective

---
## 🎉 Integration & Test

> 🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### LINKED ISSUE #129: Replace System.out.println with proper logger calls in production code
## Summary

Several production code paths use `System.out.println` instead of the Log4j 2 logger. These calls bypass all log level filtering, cannot be redirected to a log file, and in some cases are clear debug leftovers that should not be in production code at all.

## Affected Files

**Debug leftovers (should be removed or converted to DEBUG):**
- `common/src/main/java/gov/nasa/pds/registry/common/app/Handler.java:29` — `System.out.println("here")`
- `common/src/main/java/gov/nasa/pds/registry/common/dd/Pds2EsDataTypeMap.java:150` — prints entire type map on every run

**Warnings bypassing the logger:**
- `common/src/main/java/gov/nasa/pds/registry/common/connection/es/EsUtils.java:115` — `System.out.println("[WARN] " + warn)`
- `common/src/main/java/gov/nasa/pds/registry/common/connection/es/RestClientWrapper.java:61` — `System.out.println("[WARN] " + warn)`
- `common/src/main/java/gov/nasa/pds/registry/common/es/dao/dd/LddVersions.java:81–82` — prints last date and file list

**Debug utility method using println throughout:**
- `common/src/main/java/gov/nasa/pds/registry/common/util/xml/XmlNamespaces.java:28–56` — entire method uses println

## Acceptance Criteria

- No `System.out.println` or `System.err.println` calls remain in non-CLI production code paths
- Debug leftovers are removed entirely
- WARN-level messages use `log.warn(...)`
- Debug utility methods use `log.debug(...)`
- Intentional CLI output (usage text, version strings) is explicitly excluded from this change

---
## 🎉 Integration & Test

> 🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### LINKED ISSUE #130: Standardize logger field declarations across all classes
## Summary

Logger fields are declared inconsistently across the codebase: three different variable names (`log`, `LOG`, `logger`), three different instantiation scopes (static final, instance field, local variable), and mixed factory argument styles (`ClassName.class` vs `this.getClass()`). This makes the codebase harder to read and review.

## Details

**Variable name inconsistencies:**
- `log` (lowercase) — most classes (~35)
- `LOG` (all-caps static) — `AuthContent.java`, `XPathCacheLoader.java`
- `logger` — `MappingImpl.java`

**Instantiation scope issues:**
- `private static final Logger log` — correct, used in ~4 classes
- Instance field initialized in constructor with `this.getClass()` — used in ~30 classes (wasteful, creates per-instance logger objects)
- Local variable inside methods — `CloseUtils.java`, `NsUtils.java`, `LoadDDCmd.java`, both `RegistryManager` classes (anti-pattern: logger not reused across calls)

## Acceptance Criteria

- All logger fields use the standard form: `private static final Logger log = LogManager.getLogger(ClassName.class)`
- Variable name is `log` across all classes
- No instance-field or local-variable loggers remain
- Applies to all classes in `common`, `harvest`, and `manager` modules

---
## 🎉 Integration & Test

> 🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### LINKED ISSUE #131: Convert string-concatenation log calls to parameterized form
## Summary

Approximately 73 log call sites across the codebase use Java string concatenation (`"msg: " + variable`) instead of Log4j 2's parameterized form (`"msg: {}", variable`). This means the string is always constructed even when the log level is disabled, wasting CPU and memory on every suppressed log call.

## Top Offenders

- `harvest/src/main/java/gov/nasa/pds/harvest/cmd/HarvestCmd.java` — 7 sites
- `common/src/main/java/gov/nasa/pds/registry/common/es/service/SchemaUpdater.java` — 6 sites
- `common/src/main/java/gov/nasa/pds/registry/common/es/service/ProductService.java` — 6 sites
- `common/src/main/java/gov/nasa/pds/registry/common/es/dao/DataLoader.java` — 6 sites
- `manager/src/main/java/gov/nasa/pds/registry/mgr/srv/IndexService.java` — 5 sites

## Acceptance Criteria

- All log call sites use parameterized form: `log.info("key={}", value)` not `log.info("key=" + value)`
- Applies to all levels (DEBUG, INFO, WARN, ERROR) across all three modules
- No functional behavior changes — this is a pure performance/style fix

---
## 🎉 Integration & Test

> 🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### LINKED ISSUE #132: Align slf4j-nop version across harvest and manager modules
## Summary

The harvest and manager modules both declare a `slf4j-nop` dependency to silence SLF4J noise from transitive dependencies, but they pin different versions: harvest uses `2.0.17` while manager uses `2.0.16`. As sibling modules in the same Maven monorepo, they should pin the same version, ideally managed from the parent pom.

## Details

- `harvest/pom.xml` — `slf4j-nop:2.0.17`
- `manager/pom.xml` — `slf4j-nop:2.0.16`

## Acceptance Criteria

- Both modules use the same `slf4j-nop` version
- Version is managed in the parent `pom.xml` `<dependencyManagement>` section so future updates only require one change
- Chosen version is the latest stable `2.0.x` release at the time of the fix

---
## 🎉 Integration & Test

> 🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### PR DESCRIPTION
**Pull request by al-niessner**
## Summary

Comprehensive logging standardization across the `common`, `harvest`, and `manager` modules, addressing a backlog of logging issues and adding operator-facing outcome logging:

- **Consolidated `Log4jConfigurator`** from harvest + manager into `common` with a shared `LogLevels` class (includes `SUMMARY`, `LABEL_SUCCESS`, `LABEL_FAILURE`, `LABEL_SKIPPED` levels)
- **Per-product outcome logging** (SUCCESS / FAILED / SKIPPED / MATCHED) emitted per LIDVID so operators can grep logs by outcome
- **Fixed exception logging** to always pass the exception object so full stack traces are preserved
- **Replaced `System.out.println`** calls in production code with proper `log.warn`/`log.debug` calls
- **Standardized logger field declarations** across all classes to `private static final Logger log = LogManager.getLogger(ClassName.class)`; eliminates per-instance loggers and local-variable loggers
- **Converted string-concatenation log calls** to Log4j 2 parameterized form (`"msg: {}"`, `var`) across all three modules
- **Aligned `slf4j-nop` version** in parent pom so both modules use the same version
- **Removed dead code**: legacy Elasticsearch (`es/`) client classes, `UpdateAltIdsCmd`, and related test infrastructure

### AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance
- [x] AI used for moderate content generation
- [ ] AI generated substantial portions of this code

Estimated % of code influenced by AI: _40_ %

## Related Issues
Fixes #126
Fixes #127
Fixes #128
Fixes #129
Fixes #130
Fixes #131
Fixes #132

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified no sensitive information is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines.

### Testing & Validation
- [ ] **Test Accuracy:** Verified test data is accurate and representative.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** Successfully built and ran the changes locally.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed no breaking changes (or clearly documented).

### CODE REVIEW

**Review by nutjob4life — COMMENTED** [review]
I'm getting a compilation error when I try this patch:
```tcsh
mirasol 283 % cd /tmp
/tmp
mirasol 284 % git clone --branch super_log --quiet git@github.com:NASA-PDS/registry-loader.git
mirasol 285 % cd registry-loader/
/tmp/registry-loader
mirasol 286 % mvn --quiet clean test
                          null  -->  null
                                -->  null
          2013-10-24T00:00:00Z  -->  2013-10-24T00:00:00Z
      2013-10-24T00:49:37.457Z  -->  2013-10-24T00:49:37.457Z
         2013-302T01:02:03.123  -->  2013-10-29T01:02:03.123Z
        2013-302T01:02:03.123Z  -->  2013-10-29T01:02:03.123Z
                   2016-09-08Z  -->  2016-09-08T00:00:00Z
                    2013-03-02  -->  2013-03-02T00:00:00Z
                       2013-12  -->  2013-12-01T00:00:00Z
                          2013  -->  2013-01-01T00:00:00Z
                         2015Z  -->  2015-01-01T00:00:00Z
                      2013-001  -->  2013-01-01T00:00:00Z
[ERROR] COMPILATION ERROR : 
[ERROR] /private/tmp/registry-loader/manager/src/main/java/gov/nasa/pds/registry/mgr/dao/RegistryDao.java:[61,42] cannot find symbol
  symbol:   method altIds()
  location: interface gov.nasa.pds.registry.common.Response.Search
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project registry-manager: Compilation failure
[ERROR] /private/tmp/registry-loader/manager/src/main/java/gov/nasa/pds/registry/mgr/dao/RegistryDao.java:[61,42] cannot find symbol
[ERROR]   symbol:   method altIds()
[ERROR]   location: interface gov.nasa.pds.registry.common.Response.Search
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :registry-manager
48.925u 7.000s 0:29.19 191.5%	0+0k 0+0io 16pf+0w
```
Maybe I have an out-of-date dependency?

File: common/src/main/java/gov/nasa/pds/registry/common/connection/aws/SearchRespWrap.java [hunk 1]
```diff
@@ -31,12 +29,6 @@ class SearchRespWrap implements Response.Search {
     this.parent = parent;
   }
   @Override
-  public Map<String, Set<String>> altIds() throws UnsupportedOperationException, IOException {
```
Comment (al-niessner) [inline common/src/main/java/gov/nasa/pds/registry/common/connection/aws/SearchRespWrap.java:34]: @jordanpadams @nutjob4life @tloubrieu-jpl 

I will leave it to your decision, but I would remove it since it has never been used.

It all starts here. Look at the code removed and you will that it was never implemented. Hence, it could never have been called. So, I removed it.

What I missed, is that it is used in manager here:
https://github.com/NASA-PDS/registry-loader/blob/f08dacc345bcf776d3a33db7fec8be169c93fb71/manager/src/main/java/gov/nasa/pds/registry/mgr/dao/RegistryDao.java#L53-L62

Which is part of the command:
[update-alt-ids](https://github.com/NASA-PDS/registry-loader/blob/super_log/manager/src/main/java/gov/nasa/pds/registry/mgr/cmd/data/UpdateAltIdsCmd.java)

Been this way a good 2 years. If it has been ignored for 2 years do we really want to drag around an unimplemented feature that nobody is using?

Comment (jordanpadams) [reply common/src/main/java/gov/nasa/pds/registry/common/connection/aws/SearchRespWrap.java:34]: @al-niessner I don't even know what that does. Let's clean it up for now.

Are we sure that method is not called by anything else?

Comment (nutjob4life) [reply common/src/main/java/gov/nasa/pds/registry/common/connection/aws/SearchRespWrap.java:34]: > If it has been ignored for 2 years do we really want to drag around an unimplemented feature that nobody is using?

I'm with @jordanpadams on this; dropping it seems like a good approach. (And that's also _one_ way to find out if someone actually _was_ using it 😉)

**Comment by al-niessner** [comment]
@nutjob4life 

Well, the build works!

**Review by jordanpadams — APPROVED** [review]
Integration tests fail which is expected at this time

**Comment by jordanpadams** [comment]
Thanks for the work here — the direction is right. A few gaps and one bug to address before merging:

---

### #127 — Bug: `log.error` format string drops the error message

`DataLoader.java` line 352:
```java
log.error("LIDVID = {}, Message = ", sanitizedLidvid, sanitizedMessage);
```
The format string has only one `{}` but two parameters. Log4j will treat `sanitizedMessage` as the `Throwable` cause, so the actual error message is silently dropped from the log output. Should be:
```java
log.error("LIDVID = {}, Message = {}", sanitizedLidvid, sanitizedMessage);
```

---

### #129 — Two `System.out.println` sites still remain

- `common/.../util/xml/XmlNamespaces.java` — the entire `debug()` method still uses `System.out.println` throughout (lines 28–56). Per the issue AC, debug utility methods should use `log.debug(...)`.
- `common/.../es/dao/dd/LddVersions.java` lines 81–82 — `System.out.println("Last date: ...")` and `System.out.println("Files: ...")` are still present.

---

### #132 — `slf4j-nop` version is aligned but not centralized

Both modules are now on `2.0.18`, which satisfies the "same version" part. However, the issue AC specifically requires the version to be managed in the **parent `pom.xml` `<dependencyManagement>` section** so future updates only need one change. That step wasn't done.

---

### #126 — Per-product outcome logging incomplete

Three gaps against the acceptance criteria:

1. **No `LABEL_SUCCESS` in the happy path**: When `resp.errors()` is `false` (all documents accepted in a bulk call), `todo.clear()` is called but no per-product outcome line is emitted. Operators will see silence for successfully loaded products on that code path.

2. **Log format doesn't match the AC**: The issue requires a grep-friendly format, e.g.:
   ```
   [SUCCESS] lid=urn:nasa:pds:... vid=1.0
   ```
   Currently the lidvid string is logged as the bare message. The custom level name appears as the log level field in the appender pattern, not as a structured prefix in the message body. Operators can't grep for `[SUCCESS]` or `lid=`/`vid=` tokens.

3. **Manager module not instrumented**: The AC says "both harvest and manager emit outcome lines." Only harvest has been wired up.

> 🤖 Generated with help of [Claude Code](https://claude.ai/claude-code)

**Review by jordanpadams — CHANGES_REQUESTED** [review]
See comment

**Comment by nutjob4life** [comment]
@al-niessner  wrote:

> Well, the build works!

Woo-hoo! 
cat-thumbs-up

**Comment by al-niessner** [comment]
@jordanpadams 

Hey Mr Ai,

Yes to #129. See note on ticket. It states these are used by test to see private parts of class. Good as is. If tests are unused/unwanted that is something different.

#132 read the title only.
#127 ooops.
#126 ooops.

**Comment by al-niessner** [comment]
Still builds!

**Comment by jordanpadams** [comment]
@al-niessner @nutjob4life rolled in additional updates for https://github.com/NASA-PDS/registry-loader/issues/130 and https://github.com/NASA-PDS/registry-loader/issues/131

File: harvest/src/main/java/gov/nasa/pds/harvest/cmd/HarvestCmd.java [hunk 2]
```diff
@@ -289,19 +288,20 @@ private void printSummary()
         log.log(LogUtils.LEVEL_SUMMARY, "Summary:");
         int processedCount = counter.prodCounters.getTotal();
         
-        log.log(LogUtils.LEVEL_SUMMARY, "Skipped files: " + counter.skippedFileCount);
-        log.log(LogUtils.LEVEL_SUMMARY, "Loaded files: " + processedCount);
+        log.log(LogUtils.LEVEL_SUMMARY, "Label Failure: {} (database rejected label/product)", counter.failedFileCount);
+        log.log(LogUtils.LEVEL_SUMMARY, "Label Ignored: {} (files that are not readable as label/product)", counter.ignoredFileCount);
+        log.log(LogUtils.LEVEL_SUMMARY, "Label Matched: {} (batch rejected as duplicate lidvid)", counter.matchedFileCount);
```
Comment (nutjob4life) [inline harvest/src/main/java/gov/nasa/pds/harvest/cmd/HarvestCmd.java:293]: I think this will always say:

    Label Matched: 0 (batch rejected as duplicate lidvid)

Because `DataLoader.loadBatch()` now takes `Set<String> matchedIds` and `MetadataWriter` expects it to drive:
```java
counter.matchedFileCount += matchedIds.size();
```
But when the code encounters an HTTP code `409` (which, by the way, could be replaced with `HttpURLConnection.HTTP_CONFLICT`) , it only does:
```java
log.log(LogLevels.LABEL_MATCHED, sanitizedLidvid);
todo.remove(asKey(item));
```
There is no counterpart code that does something like
```java
if (matchedIds != null) matchedIds.add(item.id());
```
so the new summary ends up always being zero.

If that logic makes sense, then I'd consider this a blocker.

File: harvest/src/main/java/gov/nasa/pds/harvest/dao/MetadataWriter.java [hunk 3]
```diff
@@ -94,26 +92,29 @@ private void writeBatch() throws Exception {
...
     counter.failedFileCount += failedIds.size();
+    counter.matchedFileCount += matchedIds.size();
 
     // Update product counters
     for (RegistryDocBatch.NJsonItem item : docBatch.getItems()) {
       if ((nonRegisteredIds == null && !failedIds.contains(item.lidvid))
           || (nonRegisteredIds != null && nonRegisteredIds.contains(item.lidvid)
               && !failedIds.contains(item.lidvid))) {
         counter.prodCounters.inc(item.prodClass);
+        log.log(LogLevels.LABEL_SUCCESS, item.lidvid);
```
Comment (nutjob4life) [inline harvest/src/main/java/gov/nasa/pds/harvest/dao/MetadataWriter.java:117]: While this handles the `!failedIds.contains(item.lidvid)` case, where is the `!matchedIds.contains(item.lidvid)` case? Even after fixing the "blocker" described above for `Label Matched: 0`, a duplicate rejected with HTTP CONFLICT ("409"), can still fall through this code and get

    LABEL MATCHED urn:...
    LABEL SUCCESS urn:...

and increment the product-success counter.

Depending on the intended semantics of `nonRegisteredIds`, this should be
```java
if (nonRegisteredIds == null
    || (nonRegisteredIds.contains(item.lidvid)
        && !failedIds.contains(item.lidvid)
        && !matchedIds.contains(item.lidvid))) {
```
More generally, there are 5 states, right?

1. FAILURE
2. IGNORED
3. MATCHED
4. SKIPPED
5. SUCCESS

And they're all mutually exclusive. The implementation should enforce that as an invariant rather than deriving each counter independently.

**Review by nutjob4life — CHANGES_REQUESTED** [review]
An artificial intelligence assisted with the review (this is was a _huge_ PR, 81 files), and we came up with a couple issues (see comments below).

File: common/src/main/java/gov/nasa/pds/registry/common/util/log/LogLevels.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.apache.logging.log4j.Level;
+
+public class LogLevels {
+  private LogLevels() {}
+  public static final Level LABEL_FAILURE = Level.forName("LABEL FAILURE", 121); // batch did not accept lidvid
+  public static final Level LABEL_IGNORED = Level.forName("LABEL IGNORED", 122); // file could not be read as a label
+  public static final Level LABEL_MATCHED = Level.forName("LABEL MATCHED", 123); // batch rejected as duplicate lidvid
+  public static final Level LABEL_SKIPPED = Level.forName("LABEL SKIPPED", 124); // the lidvid was already in database
+  public static final Level LABEL_SUCCESS = Level.forName("LABEL SUCCESS", 125); // batch accepted lidvid
```
Comment (nutjob4life) [inline common/src/main/java/gov/nasa/pds/registry/common/util/log/LogLevels.java:11]: These levels are 121…125, but the Log4j standard levels are roughly

    FATAL 100
    ERROR 200
    WARN 300
    INFO 400
    DEBUG 500
    TRACE 600

So all the `LABEL_*` levels aren't fatal, but are **more severe than ERROR**.

If someone runs with

    -v ERROR

they'll still get

    LABEL SUCCESS
    LABEL SKIPPED
    …
    LABEL FAILURE

and likewith with `WARN`. If this is deliberate, great, and ignore this comment.

**Review by nutjob4life — CHANGES_REQUESTED** [review]

File: common/src/main/java/gov/nasa/pds/registry/common/util/log/LogLevels.java (hunk 4 above)
Comment (jordanpadams) [reply common/src/main/java/gov/nasa/pds/registry/common/util/log/LogLevels.java:11]: @al-niessner ☝️

Comment (al-niessner) [reply common/src/main/java/gov/nasa/pds/registry/common/util/log/LogLevels.java:11]: Just expanded what was there, but think the idea is that 100 makes them non-filtered. They will still show with -v ERROR because it works on being less than 200. I guess you could turn them off -v FATAL. Anyway, just extended what was there. If the user does -v FATAL then they deserve not see the summary.

File: harvest/src/main/java/gov/nasa/pds/harvest/dao/MetadataWriter.java (hunk 3 above)
Comment (al-niessner) [reply harvest/src/main/java/gov/nasa/pds/harvest/dao/MetadataWriter.java:117]: There are 5 states but your bot needs to look around to see them all. Since we want clear counting, ignored, matched, and skipped are all similar quasi pass or fail. See summary comments to their difference. All 5 states are accounted for but lumping them would contravene the clarity requirement/request.

File: harvest/src/main/java/gov/nasa/pds/harvest/cmd/HarvestCmd.java (hunk 2 above)
Comment (al-niessner) [reply harvest/src/main/java/gov/nasa/pds/harvest/cmd/HarvestCmd.java:293]: You want matched to always be 0. There is a race condition from latency with AOSS or there is the case of two harvests over same data at same time problem. harvest checks and thinks lidvid needs to be inserted and uploads it. Mean while already in place batch or other process has already injected it so this harvest gets an error from opensearch that lidvid matches. hence the count.

Scoff if you like. The check would not be there if we had not tripped over at least once. So either/both of these conditions exist in the real world.

**Comment by jordanpadams** [comment]
@al-niessner I hate to be a pain here, but can you attach what an example output log would actually look like so I can verify these meet the requirement expectations? Or provide examples of each? It may just be easier to update the end user documentation at https://github.com/NASA-PDS/registry regarding logging

**Comment by al-niessner** [comment]
> @al-niessner I hate to be a pain here, but can you attach what an example output log would actually look like so I can verify these meet the requirement expectations? Or provide examples of each? It may just be easier to update the end user documentation at https://github.com/NASA-PDS/registry regarding logging

Happy to update, but what files actually need editing? Not sure how things are built and did not see anything specific in root readme as I scrolled through it in 5 seconds nor a file that said set-user-expectation.edit.me or logging.edit.me or anything else.

**Comment by jordanpadams** [comment]
@al-niessner nothing needs updating, I just don't know what an output log would look like from a user perspective when harvest is run. I need that to determine if this is what we want.

**Comment by al-niessner** [comment]
@jordanpadams 

Did a small run so here is the full log file:
```
2026-08-26T16:48:10.644815353Z main SUMMARY Reading configuration from /home/niessner/Projects/PDS/registry-ref-data/custom-datasets/naif3.mt.xml
2026-08-26T16:48:10.838893086Z main SUMMARY Output directory: /tmp/harvest/out
2026-08-26T16:48:11.552304964Z main SUMMARY Connection: gov.nasa.pds.registry.common.connection.UseOpensearchSDK2@3c017078
2026-08-26T16:49:02.349145600Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice::1.0
2026-08-26T16:49:02.349443177Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice::2.0
2026-08-26T16:49:02.349760302Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice::3.0
2026-08-26T16:49:02.350021671Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:document::1.0
2026-08-26T16:49:02.350210314Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels::1.0
2026-08-26T16:49:02.350506902Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels::3.0
2026-08-26T16:49:02.350798163Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels::2.0
2026-08-26T16:49:02.351012534Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:document:spiceds::1.0
2026-08-26T16:49:02.351336447Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:ck_m2020_surf_rover_tlm_0000_0089_v1.bc::1.0
2026-08-26T16:49:02.351616514Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:ck_m2020_surf_rover_tlm_0089_0179_v1.bc::1.0
2026-08-26T16:49:02.351799242Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:ck_m2020_surf_rover_tlm_0179_0299_v1.bc::1.0
2026-08-26T16:49:02.351960533Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:mk_m2020::1.0
2026-08-26T16:49:02.352145Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:mk_m2020::2.0
2026-08-26T16:49:02.352317994Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:mk_m2020::3.0
2026-08-26T16:49:02.352488589Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:sclk_m2020_168_sclkscet_00007.tsc::1.0
2026-08-26T16:49:02.352663429Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:sclk_m2020_168_sclkscet_refit_v01.tsc::1.0
2026-08-26T16:49:02.352848670Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:sclk_m2020_168_sclkscet_refit_v02.tsc::1.0
2026-08-26T16:49:02.353054647Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:sclk_m2020_168_sclkscet_refit_v03.tsc::1.0
2026-08-26T16:49:02.353232979Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:ck_m2020_surf_ra_tlmres_0000_0089_v1.bc::1.0
2026-08-26T16:49:02.353384761Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:ck_m2020_surf_ra_tlmres_0089_0179_v1.bc::1.0
2026-08-26T16:49:02.353546099Z main LABEL SUCCESS urn:nasa:pds:mars2020.spice:spice_kernels:ck_m2020_surf_ra_tlmres_0179_0299_v1.bc::1.0
2026-08-26T16:49:02.353804730Z main SUMMARY Summary:
2026-08-26T16:49:02.353998564Z main SUMMARY Label Failure: 0 (database rejected label/product)
2026-08-26T16:49:02.354157598Z main SUMMARY Label Ignored: 0 (files that are not readable as label/product)
2026-08-26T16:49:02.354306748Z main SUMMARY Label Matched: 0 (batch rejected as duplicate lidvid)
2026-08-26T16:49:02.354477025Z main SUMMARY Label Skipped: 0 (lidvids that already existed in database)
2026-08-26T16:49:02.354632606Z main SUMMARY Label Success: 21 (database accepted label/product)
2026-08-26T16:49:02.354891025Z main SUMMARY   Product_Bundle: 3
2026-08-26T16:49:02.355022655Z main SUMMARY   Product_Collection: 4
2026-08-26T16:49:02.355158856Z main SUMMARY   Product_Document: 1
2026-08-26T16:49:02.355284220Z main SUMMARY   Product_SPICE_Kernel: 13
2026-08-26T16:49:02.355405307Z main SUMMARY Package ID: 5b1064e3-dbdf-46c6-bce4-3585a823ac56
```

Are Product_Document counted? Looks like not. Maybe there are 2 versions of something so success is 1 less than total?

**Comment by jordanpadams** [comment]
@al-niessner all labels (products) should be counted. the product type is irrelevant.

**Comment by al-niessner** [comment]
@jordanpadams 

Well, you can make this ticket last indefinitely but having me chase down the count discrepancy again, or call it done and create a new ticket for counting. Fixing the counting is a repetitive chore. I would prefer to do it with the test harness rather than ad-hoc in this ticket though.

**Comment by al-niessner** [comment]
@jordanpadams I count 3 bundles, 3 collections, 1 document, and 13 kernels in the harvest (easy, short log). Seems collections is off by 1.
